xtask/codegen/css.ungram

```diff
@@ -635,85 +613,160 @@ CssKeyframesSelectorList =  (CssKeyframesSelector (',' CssKeyframesSelector)*)
+// <mf-eq> = '='
+// <mf-comparison> = <mf-lt> | <mf-gt> | <mf-eq>
+//
+// <general-enclosed> branch must only be chosen if the input does not match either of the preceding branches.
+// <general-enclosed> exists to allow for future expansion of the grammar in a reasonably compatible way.
+// <general-enclosed> = [ <function-token> <any-value>? ) ] | [ ( <any-value>? ) ]
 
 // @media screen, all, print {}
 CssMediaAtRule =
 	'media'
 	query_list: CssMediaQueryList
-	'{'
-	body: AnyCssRule
-	'}'
+	block: CssBlock
 
+CssMediaQueryList = (AnyCssMediaQuery (',' AnyCssMediaQuery)*)
 
-// NB: a query list can't start with `and` condition
-CssMediaQueryList = (CssMediaQuery (',' CssMediaQuery)*)
+AnyCssMediaQuery =
+	CssMediaConditionQuery
+	| AnyCssMediaTypeQuery
+	| CssBogusMediaQuery
 
-// @media or screen and print {}
-// 		  ^^^^^^^^^^^^^^^^^^^^
-CssMediaQuery =
-	condition: 'not' | 'or' | 'only'?
-	type: AnyCssMediaQueryType
-	consequent: CssMediaQueryConsequent?
+CssMediaConditionQuery =
+	condition: AnyCssMediaCondition
 
+AnyCssMediaTypeQuery =
+	CssMediaAndTypeQuery
+	| CssMediaTypeQuery
 
-// @media not screen and print {}
-// 		  ^^^^^^^^^
-CssMediaQueryConsequent =
+// [ not | only ]? <media-type> [ and <media-condition-without-or> ]?
+// @media not all and not (color)  { }
+// @media not all and (color) and (color)  { }
+CssMediaAndTypeQuery =
+	left: CssMediaTypeQuery
 	'and'
-	condition: 'not'?
-	type: AnyCssMediaQueryType
+	right: AnyCssMediaTypeCondition
 
+CssMediaTypeQuery =
+	modifier: ('only' | 'not')?
+	type: CssMediaType
 
-AnyCssMediaQueryType =
-	CssIdentifier
-	| CssMediaQueryFeature
+CssMediaType =
+	value: CssIdentifier
+
+AnyCssMediaTypeCondition =
+	CssMediaNotCondition
+	| CssMediaAndCondition
+	| AnyCssMediaInParens
+
+// <media-condition> = <media-not> | <media-in-parens> [ <media-and>* | <media-or>* ]
+AnyCssMediaCondition =
+	CssMediaNotCondition
+	| CssMediaAndCondition
+	| CssMediaOrCondition
+	| AnyCssMediaInParens
+
+CssMediaNotCondition =
+	'not'
+	condition: AnyCssMediaInParens
+
+CssMediaOrCondition =
+	left: AnyCssMediaInParens
+	'or'
+	right: AnyCssMediaOrCombinableCondition
+
+AnyCssMediaOrCombinableCondition =
+	CssMediaOrCondition
+	| AnyCssMediaInParens
 
-// @media (width: 800px) {}
-// 				^^^^^^^^^^^^^^
-// @media (width) {}
-// 				^^^^^^^
-CssMediaQueryFeature =
+CssMediaAndCondition =
+	left: AnyCssMediaInParens
+	'and'
+	right: AnyCssMediaAndCombinableCondition
+
+AnyCssMediaAndCombinableCondition =
+	CssMediaAndCondition
+	| AnyCssMediaInParens
+
+AnyCssMediaInParens =
+	CssMediaConditionInParens
+	| CssMediaFeatureInParens
+
+CssMediaConditionInParens =
 	'('
-	feature: AnyCssMediaQueryFeatureType
+	condition: AnyCssMediaCondition
 	')'
 
-AnyCssMediaQueryFeatureType =
-	CssMediaQueryFeaturePlain
-	| CssMediaQueryFeatureBoolean
-	| CssMediaQueryFeatureCompare
-	| CssMediaQueryFeatureRange
+CssMediaFeatureInParens =
+	'('
+	feature: AnyCssQueryFeature
+	')'
 
+AnyCssQueryFeature =
```

Comment:
Reuse nodes for a media and a container at rule.
https://github.com/biomejs/biome/pull/1114#discussion_r1419732285